### PR TITLE
Adjust package.json fields

### PR DESCRIPTION
**Pull request by marvin-j97**
- Moved types to dev dependencies
- Added typescript as dev dependency
- Add types field
- Fixed entry points

### About the entry point
The dist folder is indeed gitignored, however, when you build for release, you want to include that dist folder, as only the JS is executable (for node). The .d.ts files are not valid entry points, but add Typescript information to the build without it staying a .ts file.

### CODE REVIEW

**Comment by marvin-j97** [comment]
I would advise against using .npmignore and instead use the [files](https://docs.npmjs.com/cli/v8/configuring-npm/package-json#files) field explicitly in the package.json, by the way

File: package-lock.json [hunk 1]
```diff
@@ -5,12 +5,9 @@
     "requires": true,
     "packages": {
         "": {
-            "name": "cubecubed",
```
Comment (imaphatduc) [inline package-lock.json:8]: I'm not sure about `package-lock.json` auto-generation process, but why is this line removed?

File: package-lock.json [hunk 2]
```diff
@@ -5,12 +5,9 @@
...
             "version": "1.0.8",
             "license": "MIT",
             "dependencies": {
-                "@types/d3": "^7.1.0",
-                "@types/katex": "^0.11.1",
                 "d3": "^7.1.1",
                 "inquirer": "^8.2.0",
                 "katex": "^0.15.1"
```
Comment (imaphatduc) [inline package-lock.json:13]: I have some problems with `katex` npm package (text rendering for example), and when I included [these lines] (https://katex.org/docs/browser.html) in `index.html`, those problems disappear. I'm considering to remove it. What's your idea about this?

**Comment by imaphatduc** [comment]
Complete merging your PR. Thank you!
